src/ERC6538Registry.sol

```diff
@@ -1,45 +1,134 @@
-// SPDX-License-Identifier: UNLICENSED
-pragma solidity 0.8.20;
-
-import {IERC6538Registry} from "./interfaces/IERC6538Registry.sol";
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.23;
 
 /// @dev `ERC6538Registry` contract to map accounts to their stealth meta-address. See
 /// [ERC-6538](https://eips.ethereum.org/EIPS/eip-6538) to learn more.
-contract ERC6538Registry is IERC6538Registry {
-  /// @notice Maps a registrant's identifier to the scheme ID to the stealth meta-address.
+contract ERC6538Registry {
+  /// @notice Next nonce expected from `user` to use when signing for `registerKeysOnBehalf`.
   /// @dev `registrant` may be a standard 160-bit address or any other identifier.
   /// @dev `schemeId` is an integer identifier for the stealth address scheme.
-  mapping(bytes registrant => mapping(uint256 schemeId => bytes)) public stealthMetaAddressOf;
+  mapping(address registrant => mapping(uint256 schemeId => bytes stealthMetaAddress)) public
+    stealthMetaAddressOf;
+
+  /// @notice A nonce used to ensure a signature can only be used once.
+  /// @dev `user` is the registrant address.
+  /// @dev `nonce` will be incremented after each valid `registerKeysOnBehalf` call.
+  mapping(address user => uint256 nonce) public nonceOf;
```

Comment:
similar style nit for consistency, I personally never use the final named param since it's redundant with the mapping var name
```suggestion
  mapping(address user => uint256) public nonceOf;
```

Reply:
Removed in [7cfeddc](https://github.com/ScopeLift/stealth-address-erc-contracts/pull/5/commits/7cfeddc3ebfcb4c4096e192d5c6ac1886e321a57)